### DISCUSSION TITLE
K1 users sometimes get a layer shift during a pause/resume of any kind. I have figured out why.

### DISCUSSION THREAD

**Discussion by grrminator** [Guilouz/Creality-Helper-Script-Wiki#696] [Issues]
When running out of filament or pausing the print for any reason, when resuming, sometimes I would see a layer shift. My printer is sound and finely tuned, and I dont have slips at any other time. ONLY when a pause and resume occurs. Layer shifts happened about 50% of the time after a pause/resume.

And I figured out why.

In the resume macro in the gcode_macros.cfg file there is a G28 X Y command in the beginning of the macro.

That line is unnecessary because the printer locks the motors and keeps them on after a pause. And it turns out that this was the cause of the layer shifting I am seeing. I made creality aware by pinging them in the reddit post I mentioned below.

I commented out that line like this: https://pastebin.com/ssiXmNZg

And it made the issue go away.

I can only imagine this is caused by current shifts/pressure when the motors has been running or possibly another property changes enough in the printer to cause that G28 command to achieve a home that is *slightly* different than the original homing at the start of the print.

I also detailed this in a reddit post to get the fix out to others who may not see it here. https://www.reddit.com/r/crealityk1/comments/1gqsdn1/read_and_fix_i_know_why_the_k1_max_layer_shifts/?rdt=34314

@Guilouz this might be worth mentioning in the helper script guide, or use the helper script to outright fix it out of the gate.

**Comment by tkunchick** [comment]
this must be something new with the latest K1 series firmware, I am running 2 or 3 versions behind and the G28 command is not in my pause macro.

````
[gcode_macro RESUME]
description: Resume the actual running print
rename_existing: RESUME_BASE
gcode:
  RESTORE_E_CURRENT
  {% if printer['gcode_macro PRINTER_PARAM'].hotend_temp|int != 0 %}
    {% if printer['gcode_macro PRINTER_PARAM'].hotend_temp|int > printer.extruder.temperature %}
      M109 S{printer['gcode_macro PRINTER_PARAM'].hotend_temp|int}
    {% else %}
      M104 S{printer['gcode_macro PRINTER_PARAM'].hotend_temp|int}
    {% endif %}
    SET_GCODE_VARIABLE MACRO=PRINTER_PARAM VARIABLE=hotend_temp VALUE=0
  {% endif %}
  {% if printer['gcode_macro PRINTER_PARAM'].fan2_speed > 0 %}
    # {% set fspeed = printer['gcode_macro PRINTER_PARAM'].fan2_speed %}
    # {action_respond_info("fan2_value2 = %s \n" % (fspeed))}
    # SET_PIN PIN=fan2 VALUE={(printer['gcode_macro PRINTER_PARAM'].fan2_speed * 255 + 0.5)|int}
    {% set s_value = (printer['gcode_macro PRINTER_PARAM'].fan2_speed * 255 - printer['gcode_macro PRINTER_PARAM'].fan2_min) * 255 / (255 - printer['gcode_macro PRINTER_PARAM'].fan2_min)|float %}
    M106 P2 S{s_value}
  {% endif %}
  {% set z_resume_move = printer['gcode_macro PRINTER_PARAM'].z_safe_pause|int %}
  {% if z_resume_move > 2 %}
    {% set z_resume_move = z_resume_move - 2 %}
    G91
    G1 Z-{z_resume_move} F600
    M400
  {% endif %}
  {action_respond_info("z_resume_move = %s"% (z_resume_move))}
  ##### read E from pause macro #####
  {% set E = printer["gcode_macro PAUSE"].extrude|float + 1.0 %}
  #### get VELOCITY parameter if specified ####
  {% if 'VELOCITY' in params|upper %}
    {% set get_params = ('VELOCITY=' + params.VELOCITY) %}
  {%else %}
    {% set get_params = "" %}
  {% endif %}
  ##### end of definitions #####
  {% if printer.extruder.can_extrude|lower == 'true' %}
    G91
    G1 E{E} F2100
    G90
  {% else %}
    {action_respond_info("Extruder not hot enough")}
  {% endif %}
  M400
  RESUME_BASE {get_params}
````

and my gcode file hasn't been touched since march.
